shared/utils/index.ts

```diff
@@ -112,3 +112,87 @@ export function checkHrefFormat(href: string): string {
 
   return href
 }
+
+export async function fetchWalletDetails(url: string): Promise<WalletAddress> {
+  const res = await fetch(url, {
+    headers: { Accept: 'application/json' }
+  })
+  if (!res.ok) {
+    if (res.status === 404) {
+      throw new WalletAddressFormatError('this wallet address does not exist')
```

Comment:
Consistent with other error messages.
```suggestion
      throw new WalletAddressFormatError('does not exist')
```